src/discof/replay/fd_replay_tile.c

```diff
@@ -2310,6 +2311,29 @@ init_after_snapshot( fd_replay_tile_ctx_t * ctx,
   curr_entry->epoch_ctx = ctx->epoch_ctx;
   ctx->epoch_forks->curr_epoch_idx = 0UL;
 
+  ulong wmark = snapshot_slot;
+  if( FD_LIKELY( wmark > fd_fseq_query( ctx->published_wmark ) ) ) {
+
+    /* The watermark has advanced likely because we loaded an
+       incremental snapshot that was downloaded just-in-time.  We had
+       kicked off repair with an older incremental snapshot, and so now
+       we have to prune the relevant data structures, so replay can
+       start from the latest frontier.
+
+       No funk_and_txncache_publish( ctx, wmark, &xid ); because there
+       are no funk txns to publish, and all rooted slots have already
+       been registered in the txncache when we loaded the snapshot. */
+
+    FD_LOG_NOTICE(( "wmk %lu => %lu", fd_fseq_query( ctx->published_wmark ), wmark ));
+    if( FD_LIKELY( ctx->blockstore ) ) fd_blockstore_publish( ctx->blockstore, ctx->blockstore_fd, wmark );
+    if( FD_LIKELY( ctx->forks ) ) fd_forks_publish( ctx->forks, wmark, ctx->ghost );
+    if( FD_LIKELY( ctx->ghost ) ) {
+      fd_epoch_forks_publish( ctx->epoch_forks, ctx->ghost, wmark );
+      fd_ghost_publish( ctx->ghost, wmark );
+    }
+    fd_fseq_update( ctx->published_wmark, wmark );
```

Comment:
we can merge this fix for now to get JIT snapshots to work, but i think it might be cleaner to not initialize these structures at all until the JIT incremental snapshot is loaded